packages/legend-query-builder/src/components/__tests__/QueryBuilderFetchStructure.test.tsx

```diff
@@ -713,7 +714,54 @@ test(
       (
         queryBuilderState.fetchStructureState
           .implementation as QueryBuilderTDSState
-      ).windowState.validationIssues?.length === 1,
+      ).windowState.windowValidationIssues?.length,
+    ).toBe(1);
+  },
+);
+
+test(
+  integrationTest(
+    'Query builder window function shows validation error if invalid column order',
+  ),
+  async () => {
+    const { queryBuilderState } = await TEST__setUpQueryBuilder(
+      TEST_DATA__ComplexRelationalModel,
+      stub_RawLambda(),
+      'model::relational::tests::simpleRelationalMapping',
+      'model::MyRuntime',
+      TEST_DATA__ModelCoverageAnalysisResult_ComplexRelational,
     );
+    await act(async () => {
+      queryBuilderState.initializeWithQuery(
+        create_RawLambda(
+          TEST_DATA_projectionWithWindowFunction.parameters,
+          TEST_DATA_projectionWithWindowFunction.body,
+        ),
+      );
+    });
+
+    const tdsState = guaranteeType(
+      queryBuilderState.fetchStructureState.implementation,
+      QueryBuilderTDSState,
+    );
+    tdsState.setShowWindowFuncPanel(true);
+
+    await act(async () => {
+      tdsState.windowState.moveColumn(1, 0);
+    });
+
+    expect(
```

Comment:
can we do UI tests more than state tests. 
Can we load maybe an invlaid query and assert we have the error banned on the window function ?